src/main/kotlin/io/github/mojira/arisa/modules/RemoveNonStaffTagsModule.kt

```diff
@@ -0,0 +1,53 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import arrow.syntax.function.partially1
+import io.github.mojira.arisa.domain.Comment
+import io.github.mojira.arisa.domain.Issue
+import java.time.Instant
+
+class RemoveNonStaffTagsModule(private val removalReason: String, private val removePrefix: String) : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val updateMeqsComments = comments
+                .filter(::hasMeqsTag)
+                .filter(::isNotStaffRestricted)
+                .map { it.restrict.partially1(removeTags(it.body!!, """(MEQS)(_[A-Z_]+)""".toRegex())) }
+            val updatePrefixedComments = comments
+                .filter(::hasPrefixedTag)
+                .filter(::isNotVolunteerRestricted)
+                .map { it.restrict.partially1(removeTags(it.body!!, """($removePrefix)(_[A-Z_]+)""".toRegex())) }
```

Comment:
Maybe to prevent code duplication for `has...()` and `removeTags()`, just call `removeTags` and then filter all out all comment bodies where removing tags had no effect. There there would be no need for a separate `has...()` function.

Reply:
I've refactored the module to include the list of prefixes, but I don't think it's necessary to remove the `hasPrefixedTag()` function, however in the refactor there is only a single function for this.